src/Microsoft.SqlTools.ServiceLayer/Agent/JobUtilities.cs

```diff
@@ -99,8 +109,21 @@ public static AgentJobHistoryInfo ConvertToAgentJobHistoryInfo(DataRow jobRow, S
             jobHistoryInfo.OperatorPaged = operatorPaged;
             jobHistoryInfo.RetriesAttempted = retriesAttempted;
             jobHistoryInfo.Server = server;
-
             return jobHistoryInfo;
         }
+
+        public static AgentJobStep ConvertToAgentJobStep(DataRow jobRow, SqlConnectionInfo sqlConnInfo)
+        {
+            int stepId = Convert.ToInt32(jobRow[UrnStepID], System.Globalization.CultureInfo.InvariantCulture);
+            string stepName = Convert.ToString(jobRow[UrnStepName], System.Globalization.CultureInfo.InvariantCulture);
+            string message = Convert.ToString(jobRow[UrnMessage], System.Globalization.CultureInfo.InvariantCulture);
+            DateTime runDate = Convert.ToDateTime(jobRow[UrnRunDate], System.Globalization.CultureInfo.InvariantCulture);
```

Comment:
any concerns that any of these `Convert` calls may throw an exception based on input data?  Or do we know the inputs will always be safe for some reason?

Reply:
It will always be safe as `JobHistoryItem.cs` handles and populates the fields the same way.